x/compliance/keeper/msg_server_certify_model.go

```diff
@@ -30,6 +30,17 @@ func (k msgServer) CertifyModel(goCtx context.Context, msg *types.MsgCertifyMode
 		)
 	}
 
+	// The corresponding Model Version must be present on ledger.
+	modelVersion, found := k.modelKeeper.GetModelVersion(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion)
+	if !found {
+		return nil, modeltypes.NewErrModelVersionDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	// check if softwareVersionString matches with what is stored for the given version
+	if modelVersion.SoftwareVersionString != msg.SoftwareVersionString {
+		return nil, types.NewErrModelVersionStringDoesNotMatch(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.SoftwareVersionString)
+	}
+
 	complianceInfo, found := k.GetComplianceInfo(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
 	// nolint:nestif
 	if found {
```

Comment:
Please fix the comment on line 48 `The corresponding Model Info is not required to be on the ledger`

Reply:
Fixed in scope of https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/182